MZLogic.php

```diff
@@ -270,12 +270,12 @@ function MZMoveCard($player, $search, $where, $may=false, $isReveal=false, $sile
   else AddDecisionQueue("WRITELOG", $player, "Card chosen: <0>", 1);
 }
 
-function MZChooseAndDestroy($player, $search, $may=false)
+function MZChooseAndDestroy($player, $search, $may=false, $allArsenal=false)
 {
   AddDecisionQueue("MULTIZONEINDICES", $player, $search);
   if($may) AddDecisionQueue("MAYCHOOSEMULTIZONE", $player, "<-", 1);
   else AddDecisionQueue("CHOOSEMULTIZONE", $player, "<-", 1);
-  AddDecisionQueue("MZDESTROY", $player, "-", 1);
+  AddDecisionQueue("MZDESTROY", $player, $allArsenal, 1);
```

Comment:
Why does something called MZDESTROY have a parameter called allArsenal? Ideally this should be more generic. It seems like if we're selecting a particular card from their arsenal, we shouldn't even need that parameter. We should just be destroying the particular card we selected.

Reply:
Because they printed cards after Command and Conquer that do not destroy all arsenal, so we need a parameter to make the difference. I pushed a new name to that parameter to be more generic.